cfme/fixtures/cli.py

```diff
@@ -167,18 +227,20 @@ def get_apps(appliance, old_version, count, preconfigured, pytest_config):
             preconfigured=preconfigured,
             provider_type="rhevm",
             lease_time=180,
-            version=str(usable_sorted[-1]),
+            version=str(picked_version),
         )
-        urls = cfme_data["basic_info"][update_url]
+        url = cfme_data["basic_info"][update_url]
+        assert get_puddle_cfme_version(url) == appliance.version
         for app in apps:
             app.db.extend_partition()
             app.ssh_client.run_command(
-                "curl {} -o /etc/yum.repos.d/update.repo".format(urls)
+                "curl {} -o /etc/yum.repos.d/update.repo".format(url)
             )
+
         yield apps
     except AuthException:
         msg = ('Sprout credentials key or yaml maps missing or invalid,'
-               'unable to provision appliance version %s'.format(str(usable_sorted[-1])))
+               'unable to provision appliance version %s'.format(str(picked_version)))
```

Comment:
Please use this instead 
        
          'unable to provision appliance version {}'.format(str(picked_version)
                                  or
         'unable to provision appliance version {version}'.format(str(version=picked_version)